# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/HyperTextAccessible.h
# Commit: e488a2686d0b
# Full Hash: e488a2686d0b6354c723dd896c6ce20749c8b3d7
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-12-06 03:46:09
# Regressor Bug: 1766794
# File Overlap Count: 1
# Description:
#   Bug 1802268 part 2: Reinstate partial invalidation of cached HyperText offsets for local HyperTextAccessible. r=eeejay
#   
#   In bug 1766794, I replaced partial invalidation of cached offsets with full cache invalidation.
#   I did this to simplify the code, since the offsets cache was being extended to RemoteAccessible.
#   This is fine for RemoteAccessible because offsets are only needed there when clients query them.
# ==============================================================================

diff -r 80097f40fc50 -r e488a2686d0b accessible/generic/HyperTextAccessible.h
--- a/accessible/generic/HyperTextAccessible.h	Mon Dec 05 23:27:23 2022 +0000
+++ b/accessible/generic/HyperTextAccessible.h	Mon Dec 05 23:27:24 2022 +0000
@@ -164,8 +164,6 @@
 
   virtual already_AddRefed<AccAttributes> DefaultTextAttributes() override;
 
-  virtual void InvalidateCachedHyperTextOffsets() override { mOffsets.Clear(); }
-
   // HyperTextAccessibleBase provides an overload which takes an Accessible.
   using HyperTextAccessibleBase::GetChildOffset;
 
@@ -429,10 +427,7 @@
   // HyperTextAccessibleBase
   virtual const Accessible* Acc() const override { return this; }
 
-  virtual const nsTArray<int32_t>& GetCachedHyperTextOffsets() const override {
-    if (mOffsets.IsEmpty()) {
-      BuildCachedHyperTextOffsets(mOffsets);
-    }
+  virtual nsTArray<int32_t>& GetCachedHyperTextOffsets() override {
     return mOffsets;
   }
 